OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -562,12 +553,15 @@ protected override MovePart OnComplete(Actor self, Mobile mobile, Move parent)
 
 		class MoveSecondHalf : MovePart
 		{
-			public MoveSecondHalf(Move move, WPos from, WPos to, WAngle fromFacing, WAngle toFacing, int startingFraction)
-				: base(move, from, to, fromFacing, toFacing, startingFraction) { }
+			public MoveSecondHalf(Move move, WPos from, WPos to, WAngle fromFacing, WAngle toFacing, int carryoverProgress)
+				: base(move, from, to, fromFacing, toFacing, carryoverProgress) { }
 
 			protected override MovePart OnComplete(Actor self, Mobile mobile, Move parent)
 			{
 				mobile.SetPosition(self, mobile.ToCell);
+
+				// Move might immediately queue a new MoveFirstHalf, so we need to pass a carryoverProgress there too
```

Comment:
I find this comment a bit confusing - this is the only place that sets `carryoverProgress`, but the "there too" suggests there are more.

I also found it difficult to clearly explain this, but eventually came up with this:
```csharp
// Move might immediately queue a new MoveFirstHalf within the same tick if we haven't
// reached the end of the requested path. Make sure that any leftover movement progress is
// correctly carried over into this new activity to avoid a glitch in the apparent move speed.
```